src/Diagnostics.RuntimeHost/Controllers/SitesController.cs

```diff
@@ -62,6 +63,8 @@ public SitesController(ICompilerHostClient compilerHostClient, ISourceWatcherSer
                 return BadRequest(errorMessage);
             }
 
+            string requestId = this.Request.Headers[HeaderConstants.RequestIdHeaderName];
```

Comment:
this will return null and not throw key not found exception right?

Reply:
I should do try get value. Fixing at all places